astropy/table/_column_mixins.pyx

```diff
@@ -52,6 +52,9 @@ ctypedef object (*item_getter)(object, object)
 
 
 cdef inline object base_getitem(object self, object item, item_getter getitem):
+    if (<np.ndarray>self).ndim == 0:
```

Comment:
This change may be the one causing problems...  Though see larger comment on what would be the right solution in the original issue.

Reply:
it looks my latest attempt actually works (and doesn't break anything) !